Configuration/Applications/python/ConfigBuilder.py

```diff
@@ -1665,10 +1657,11 @@ def prepare_RECOBEFMIX(self, sequence = "reconstruction"):
 
     def prepare_PAT(self, sequence = "miniAOD"):
         ''' Enrich the schedule with PAT '''
-	self.prepare_PATFILTER(self)
-        self.loadDefaultOrSpecifiedCFF(sequence,self.PATDefaultCFF,1) #this is unscheduled
-	if not self._options.runUnscheduled:	
-		raise Exception("MiniAOD production can only run in unscheduled mode, please run cmsDriver with --runUnscheduled")
+        self.prepare_PATFILTER(self)
+        self.loadDefaultOrSpecifiedCFF(sequence,self.PATDefaultCFF)
+        self.scheduleSequence(sequence.split('.')[-1], 'pat_step')
+        if not self._options.runUnscheduled:
```

Comment:
runUnscheduled is in the interface of cmsDriver as an option. I was hesitant to change its name because it is in that interface. You are right that after this PR it controls whether the convertToUnscheduled function is called or not and also the call to
"self.pythonCfgCode += self.addCustomise(1)". This brings in some other special customisation calls through the cmsDriver option --customise_unsch. I was also hesitant to change that because it is in the cmsDriver interface. I did not check to see if anything uses that cmsDriver option. I do not know what PAT does if runUnscheduled is not set, but I left the check because it was there before. My first thought is that this is involved enough that if we want to change this, it might be better in a separate PR. In an early version of my code I had actually removed the runUnscheduled variable from ConfigBuilder.py, but then I put it back when I saw those things in the cmsDriver interface. But I will include changes to this in this PR if you ask me to. I agree that the cmsDriver interface could be improved in light of the changes that are happening.